### PR TITLE
Patch for Source-Build Stage 2 FSharp FS0064 Warnings

### PR DESCRIPTION
**Pull request by ellahathaway**
Closes https://github.com/dotnet/source-build/issues/4603

Tested these changes with a local stage 2 build, and the fsharp build succeeded.

Should be reversed with https://github.com/dotnet/fsharp/issues/17713

### CODE REVIEW

**Comment by MichaelSimons** [comment]
I think we should log a separate source-build issue to track cleaning up this change once the FSharp issue flows in.

File: src/SourceBuild/content/repo-projects/fsharp.proj [hunk 1]
```diff
@@ -18,6 +18,10 @@
       custom sourcebuild arg is the one that's required. This avoids running bootstrapping twice.
     -->
     <BuildArgs Condition="'$(DotNetBuildSourceOnly)' == 'true'">$(BuildArgs) $(FlagParameterPrefix)sourcebuild $(FlagParameterPrefix)tfm $(NetCurrent) /p:SourceBuildBootstrapTfm=$(NetCurrent)</BuildArgs>
+        
+    <!-- See https://github.com/dotnet/source-build/issues/4603. 
+    Should be removed with https://github.com/dotnet/fsharp/issues/17713 -->
+    <NoWarn Condition="'$(DotNetBuildSourceOnly)' == 'true'">FS0064</NoWarn>
```
Comment (MichaelSimons) [inline src/SourceBuild/content/repo-projects/fsharp.proj:24]: Doesn't think need to flow into the inner FSharp build?  How does that work?  All of the infra to flow nowarns was removed.  I was expecting a patch was necessary to fix this.

Comment (ellahathaway) [reply src/SourceBuild/content/repo-projects/fsharp.proj:24]: Oh interesting. It passed on my local build. I'll pull the binlogs from that and double check that this flows to the inner FSharp build.

Comment (ellahathaway) [reply src/SourceBuild/content/repo-projects/fsharp.proj:24]: Ok yes, you were right, the `NoWarn` doesn't flow. I'm not sure why my local build passed, perhaps I pulled the wrong SDK or the error doesn't happen on ubuntu2404 stage 2 builds? Not sure.

Regardless, FSharp has a fix out in https://github.com/dotnet/fsharp/pull/17748 so I'll patch with that.

**Comment by ellahathaway** [comment]
Testing patch with [this pipeline run](https://dev.azure.com/dnceng/internal/_build/results?buildId=2539960&view=results) (internal Microsoft link)

**Review by MichaelSimons — APPROVED** [review]

**Pull request merged**